### PR TITLE
feat: Registration page, Razaorpay

### LINKED ISSUE #2: Registration page
- [x] Event Registrations
- [x] Payment Gateway

### LINKED ISSUE #9: Razorpay

### LINKED ISSUE #17: Email Service
- [x] Sending Emails, use `react-email`

### PR DESCRIPTION
**Pull request by rabilrbl**
resolves #9 #2  #17

Create Registration Page
Integrate RazorPay payment portal with DB

### CODE REVIEW

**Review by anishshobithps — DISMISSED** [review]
Remove Student and Faculty Model, as we are going to directly use Events from the JSON, also Events model can be removed.

**Comment by rabilrbl** [comment]
> Remove Student and Faculty Model, as we are going to directly use Events from the JSON, also Events model can be removed.

Currently we are not saving events registered by the user? How will we verify no. of events registered = amount paid?

**Comment by anishshobithps** [comment]
> > Remove Student and Faculty Model, as we are going to directly use Events from the JSON, also Events model can be removed.
> 
> Currently we are not saving events registered by the user? How will we verify no. of events registered = amount paid?

Dammit, right, **ahhhhh, stress**

**Comment by anishshobithps** [comment]
> > Remove Student and Faculty Model, as we are going to directly use Events from the JSON, also Events model can be removed.
> 
> Currently we are not saving events registered by the user? How will we verify no. of events registered = amount paid?

How about we just map the JSON file and create events in the database ? let's make a API endpoint to create events in the database ? or else we can have `seed.ts` in the `prisma/` folder and just seed the events ? Docs: https://www.prisma.io/docs/orm/prisma-migrate/workflows/seeding

**Comment by rabilrbl** [comment]
> > > Remove Student and Faculty Model, as we are going to directly use Events from the JSON, also Events model can be removed.
> > 
> > 
> > Currently we are not saving events registered by the user? How will we verify no. of events registered = amount paid?
> 
> How about we just map the JSON file and create events in the database ? let's make a API endpoint to create events in the database ? or else we can have `seed.ts` in the `prisma/` folder and just seed the events ? Docs: https://www.prisma.io/docs/orm/prisma-migrate/workflows/seeding

Seeding is better

**Comment by anishshobithps** [comment]
We might need to convert to JSON to Prisma Schema, once @melroydsa finishes it , find this 
https://github.com/omar-dulaimi/json-to-prisma-schema-convertor although it directly outputs at `prisma/schema.prisma` file we don't want that, make sure you output to a different file and copy and paste in the main schema.

File: src/lib/utils.ts [hunk 1]
```diff
@@ -10,3 +11,79 @@ export const baseURL = process.env.NEXT_PUBLIC_URL
...
+       else {
+        alert(res.message);
+       }
+    },
+  };
+  
+  const paymentObject = new window.Razorpay(options);
+  
+  paymentObject.on("payment.failed", function (response: any) {
+    alert("Payment failed. Please try again. Contact support for help"+response.error.description);
```
Comment (anishshobithps) [inline src/lib/utils.ts]: Ensure you use, `toast` or `Sonner` before pushing this

File: src/lib/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
+import { Task } from "gantt-task-react";
```
Comment (anishshobithps) [inline src/lib/helpers.ts:1]: I guess this file is not required as well as the package

File: src/components/ui/checkout.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            <li className="flex items-center justify-between font-semibold">
+              <span className="text-muted-foreground">Total</span>
+              <span>
+                {"\u20B9"}
+                {sumOfCheckedItemsAmount} /-
+              </span>
+            </li>
+          </ul>
+          <Suspense fallback={null}>
+            <Buy
```
Comment (anishshobithps) [inline src/components/ui/checkout.tsx:78]: is it required separately ? the Buy component, if possible please use the makePayment function here directly.

**Review by anishshobithps — CHANGES_REQUESTED** [review]

**Comment by anishshobithps** [comment]
Participant, Phone Number, USN should also be taken, which is not returned by the `Google Auth`

**Comment by rabilrbl** [comment]
@anishshobithps I discussed with sir
USN is not required
participants, phone number and college name
Team name for team events

File: src/lib/utils.ts (hunk 1 above)
Comment (rabilrbl) [reply src/lib/utils.ts]: yup

File: src/lib/helpers.ts (hunk 2 above)
Comment (rabilrbl) [reply src/lib/helpers.ts:1]: @joywin2003 Clean this

**Comment by rabilrbl** [comment]
This PR will not be merged

**Comment by rabilrbl** [comment]
This PR was merged in another PR created from this branch.

**Pull request closed without merging**